### PR TITLE
Upgrade wasmer version to v5.0.1

### PR DESCRIPTION
**Pull request by andreiltd**
Depends on https://github.com/wasmerio/wasmer/pull/5245

### CODE REVIEW

File: Cargo.toml
Comment (andreiltd) [inline Cargo.toml:60]: Remove this section before merging.

**Comment by jprendes** [comment]
Now that your wasmer PR gor merged, we should try with a git dependency.
Following what we did for wamr, using a git dependency for the shim binaries should be fine, as we don't publish those crates

**Comment by jprendes** [comment]
Yay, upgrading wasmer fixes the issue with bumping the rust version! @Mossaka

**Comment by jprendes** [comment]
I would be happy to use the git dependency as the actual dependency (instead of being a patch), and have it merged.
That would unblock bumping rust, and the crate is not published to a registry anyway.
We can always move back to published wasmer versions when they release a new one with the fix.

**Review by Mossaka — APPROVED** [review]

**Pull request merged**
